File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0;
        int right = heights.size() - 1;
        int currMax = 0;
        while(left < right) {
            int h;
            if(heights.at(left) < heights.at(right)) {
                h = heights.at(left);
            }
            else {
                h = heights.at(right);
            }
            int w = right - left;
            int amount = w * h;
            if (amount > currMax) {
                currMax = amount;
            }

            if (heights.at(left) < heights.at(right)) {
                left++;
            }
            else {
                right--;
            }
        }
        return currMax;
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def getNum(self, root: TreeNode, currMax: int) -> int:
            if not root:
                return 0
            newMax = currMax
            if root.val >= currMax:
                newMax = root.val
                return getNum(self, root.left, newMax) + getNum(self, root.right, newMax) + 1
            return getNum(self, root.left, newMax) + getNum(self, root.right, newMax)
        return getNum(self, root, root.val)